src/main.rs

```diff
@@ -247,6 +247,10 @@ fn perform_op(
         return ("success".to_string(), Some(new_contents));
     }
 
+    if new_contents == contents {
+        return ("success".to_string(), None);
```

Comment:
wtf is this return type??? can you open another PR to use eyre::Result? or anyhow idk if we have one or the other already